packages/core/rsc/streamable.tsx

```diff
@@ -300,6 +317,16 @@ export function render<
   messages: Parameters<
     typeof OpenAI.prototype.chat.completions.create
   >[0]['messages'];
+  /**
+   * Control how text, function calls, and tool calls are composed into a single UI.
+   *
+   * Per default, text, function and tool nodes are wrapped in a React Fragment.
+   */
+  compose?: Renderer<{
+    text: React.ReactNode | undefined;
+    functionCall: { name: keyof FS; node: React.ReactNode } | undefined;
+    toolCalls: { name: keyof TS; id: string; node: React.ReactNode }[];
```

Comment:
I'm not sure what the author of #1166 exactly had in mind, but it's possible that this might fix the issue.